File: apps/frontend/lib/workflow-service.ts
import { apiRequest } from "./api"

/**
 * 工作流接口
 */
export interface Workflow {
  id: string
  name: string
  description: string
  version: string
  definition: string
  status: string
  createdBy: string
  tags: string[]
  config: Record<string, any>
  createdAt: string
  updatedAt: string
  lastExecutedAt: string
  executionCount: number
}

/**
 * 工作流执行记录接口
 */
export interface WorkflowExecution {
  id: string
  workflowId: string
  status: string
  inputs: Record<string, any>
  outputs: Record<string, any>
  currentNodeId: string
  errorMessage: string
  startTime: string
  endTime: string
  duration: number
  retryCount: number
  executedBy: string
}

/**
 * 获取所有工作流
 */
export async function getWorkflows() {
  return apiRequest<Workflow[]>("/api/workflows")
}

/**
 * 获取工作流详情
 */
export async function getWorkflow(workflowId: string) {
  return apiRequest<Workflow>(`/api/workflows/${workflowId}`)
}

/**
 * 创建工作流
 */
export async function createWorkflow(data: {
  name: string
  description?: string
  definition: string
}) {
  return apiRequest<Workflow>("/api/workflows", {
    method: "POST",
    body: JSON.stringify(data),
  })
}

/**
 * 更新工作流
 */
export async function updateWorkflow(workflowId: string, data: Partial<Workflow>) {
  return apiRequest<Workflow>(`/api/workflows/${workflowId}`, {
    method: "PUT",
    body: JSON.stringify(data),
  })
}

/**
 * 删除工作流
 */
export async function deleteWorkflow(workflowId: string) {
  return apiRequest<void>(`/api/workflows/${workflowId}`, {
    method: "DELETE",
  })
}

/**
 * 发布工作流
 */
export async function publishWorkflow(workflowId: string) {
  return apiRequest<void>(`/api/workflows/${workflowId}/publish`, {
    method: "POST",
  })
}

/**
 * 归档工作流
 */
export async function archiveWorkflow(workflowId: string) {
  return apiRequest<void>(`/api/workflows/${workflowId}/archive`, {
    method: "POST",
  })
}

/**
 * 执行工作流
 */
export async function executeWorkflow(workflowId: string, inputs: Record<string, any> = {}) {
  return apiRequest<WorkflowExecution>(`/api/workflows/${workflowId}/execute`, {
    method: "POST",
    body: JSON.stringify({ inputs }),
  })
}

/**
 * 获取执行记录
 */
export async function getExecution(executionId: string) {
  return apiRequest<WorkflowExecution>(`/api/workflows/executions/${executionId}`)
}

/**
 * 获取工作流的所有执行记录
 */
export async function getWorkflowExecutions(workflowId: string) {
  return apiRequest<WorkflowExecution[]>(`/api/workflows/${workflowId}/executions`)
}

/**
 * 取消执行
 */
export async function cancelExecution(executionId: string) {
  return apiRequest<void>(`/api/workflows/executions/${executionId}/cancel`, {
    method: "POST",
  })
}

/**
 * 暂停执行
 */
export async function pauseExecution(executionId: string) {
  return apiRequest<void>(`/api/workflows/executions/${executionId}/pause`, {
    method: "POST",
  })
}

/**
 * 恢复执行
 */
export async function resumeExecution(executionId: string) {
  return apiRequest<void>(`/api/workflows/executions/${executionId}/resume`, {
    method: "POST",
  })
}

/**
 * 重试执行
 */
export async function retryExecution(executionId: string) {
  return apiRequest<WorkflowExecution>(`/api/workflows/executions/${executionId}/retry`, {
    method: "POST",
  })
}

/**
 * 获取执行日志
 */
export async function getExecutionLogs(executionId: string) {
  return apiRequest<string>(`/api/workflows/executions/${executionId}/logs`)
}

/**
 * 获取工作流统计信息
 */
export async function getWorkflowStatistics(workflowId: string) {
  return apiRequest<{
    totalExecutions: number
    successfulExecutions: number
    failedExecutions: number
    runningExecutions: number
    averageDuration: number
  }>(`/api/workflows/${workflowId}/statistics`)
}

// 带 Toast 的版本
export async function getWorkflowsWithToast() {
  try {
    const response = await getWorkflows()
    return response
  } catch (error) {
    throw error
  }
}

export async function createWorkflowWithToast(data: {
  name: string
  description?: string
  definition: string
}) {
  try {
    const response = await createWorkflow(data)
    return response
  } catch (error) {
    throw error
  }
}

export async function updateWorkflowWithToast(workflowId: string, data: Partial<Workflow>) {
  try {
    const response = await updateWorkflow(workflowId, data)
    return response
  } catch (error) {
    throw error
  }
}

export async function deleteWorkflowWithToast(workflowId: string) {
  try {
    const response = await deleteWorkflow(workflowId)
    return response
  } catch (error) {
    throw error
  }
}

File: apps/backend/src/main/java/org/xhy/domain/workflow/WorkflowService.java
package org.xhy.domain.workflow;

import org.xhy.domain.workflow.constant.ExecutionStatus;
import org.xhy.domain.workflow.constant.WorkflowStatus;
import org.xhy.domain.workflow.model.Workflow;
import org.xhy.domain.workflow.model.WorkflowExecution;

import java.util.List;
import java.util.Map;

/**
 * 工作流服务接口
 */
public interface WorkflowService {

    /**
     * 创建工作流
     *
     * @param workflow 工作流
     * @return 创建的工作流
     */
    Workflow createWorkflow(Workflow workflow);

    /**
     * 更新工作流
     *
     * @param workflow 工作流
     * @return 更新的工作流
     */
    Workflow updateWorkflow(Workflow workflow);

    /**
     * 删除工作流
     *
     * @param workflowId 工作流ID
     */
    void deleteWorkflow(String workflowId);

    /**
     * 获取工作流
     *
     * @param workflowId 工作流ID
     * @return 工作流
     */
    Workflow getWorkflow(String workflowId);

    /**
     * 获取所有工作流
     *
     * @return 工作流列表
     */
    List<Workflow> getAllWorkflows();

    /**
     * 按状态获取工作流
     *
     * @param status 状态
     * @return 工作流列表
     */
    List<Workflow> getWorkflowsByStatus(WorkflowStatus status);

    /**
     * 发布工作流
     *
     * @param workflowId 工作流ID
     */
    void publishWorkflow(String workflowId);

    /**
     * 归档工作流
     *
     * @param workflowId 工作流ID
     */
    void archiveWorkflow(String workflowId);

    /**
     * 执行工作流
     *
     * @param workflowId 工作流ID
     * @param inputs 输入参数
     * @return 执行记录
     */
    WorkflowExecution executeWorkflow(String workflowId, Map<String, Object> inputs);

    /**
     * 获取执行记录
     *
     * @param executionId 执行ID
     * @return 执行记录
     */
    WorkflowExecution getExecution(String executionId);

    /**
     * 获取工作流的所有执行记录
     *
     * @param workflowId 工作流ID
     * @return 执行记录列表
     */
    List<WorkflowExecution> getWorkflowExecutions(String workflowId);

    /**
     * 取消执行
     *
     * @param executionId 执行ID
     */
    void cancelExecution(String executionId);

    /**
     * 暂停执行
     *
     * @param executionId 执行ID
     */
    void pauseExecution(String executionId);

    /**
     * 恢复执行
     *
     * @param executionId 执行ID
     */
    void resumeExecution(String executionId);

    /**
     * 重试执行
     *
     * @param executionId 执行ID
     * @return 新的执行记录
     */
    WorkflowExecution retryExecution(String executionId);

    /**
     * 获取执行日志
     *
     * @param executionId 执行ID
     * @return 日志内容
     */
    String getExecutionLogs(String executionId);

    /**
     * 获取工作流统计信息
     *
     * @param workflowId 工作流ID
     * @return 统计信息
     */
    WorkflowStatistics getStatistics(String workflowId);

    /**
     * 工作流统计信息
     */
    class WorkflowStatistics {
        private final long totalExecutions;
        private final long successfulExecutions;
        private final long failedExecutions;
        private final long runningExecutions;
        private final double averageDuration;

        public WorkflowStatistics(long totalExecutions, long successfulExecutions,
                                  long failedExecutions, long runningExecutions, double averageDuration) {
            this.totalExecutions = totalExecutions;
            this.successfulExecutions = successfulExecutions;
            this.failedExecutions = failedExecutions;
            this.runningExecutions = runningExecutions;
            this.averageDuration = averageDuration;
        }

        public long getTotalExecutions() {
            return totalExecutions;
        }

        public long getSuccessfulExecutions() {
            return successfulExecutions;
        }

        public long getFailedExecutions() {
            return failedExecutions;
        }

        public long getRunningExecutions() {
            return runningExecutions;
        }

        public double getAverageDuration() {
            return averageDuration;
        }
    }
}
